patches/net/minecraft/advancements/critereon/EntityPredicate.java.patch

```diff
@@ -0,0 +1,52 @@
+--- a/net/minecraft/advancements/critereon/EntityPredicate.java
++++ b/net/minecraft/advancements/critereon/EntityPredicate.java
+@@ -36,9 +_,10 @@
+     Optional<EntityPredicate> vehicle,
+     Optional<EntityPredicate> passenger,
+     Optional<EntityPredicate> targetedEntity,
+-    Optional<String> team
++    Optional<String> team,
++    Optional<net.neoforged.neoforge.common.advancements.critereon.ICustomEntityPredicate> customLogic
+ ) {
+-    public static final Codec<EntityPredicate> CODEC = ExtraCodecs.recursive(
++    public static final Codec<EntityPredicate> VANILLA_CODEC = ExtraCodecs.recursive(
+         "EntityPredicate",
+         p_297888_ -> RecordCodecBuilder.create(
+                 p_297890_ -> p_297890_.group(
+@@ -59,8 +_,26 @@
+                         .apply(p_297890_, EntityPredicate::new)
+             )
+     );
++    public static Codec<EntityPredicate> CODEC = ExtraCodecs.<net.neoforged.neoforge.common.advancements.critereon.ICustomEntityPredicate, EntityPredicate>either(
```

Comment:
Why is this field not `final`?

Reply:
Because copied it from ItemPredicate and didn't notice. Should I also fix the patch in ItemPredicate or add a todo for 1.20.5 because it technically is a breaking change to make the codec we add in ItemPredicate final?

Reply:
Hmmm. For consistency with our stability policy, mark the field in `ItemPredicate` with a TODO for 1.20.5 to become `final`.